include/revng/Pipeline/TupleTreeInvalidationEvent.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Pipeline/InvalidationEvent.h"
+#include "revng/TupleTree/TupleTreeDiff.h"
+
+namespace pipeline {
+
+template<TupleTreeCompatible T>
+class TupleTreeInvalidationEvent
```

Comment:
Given the current design, will we ever have anything different from a `TupleTreeInvalidationEvent`?
Maybe there's a simplification opportunity here?

Reply:
not really, TupleTreeInvalidationEvent is a template, we would still need a base class, that is IvalidationEvent

Reply:
I see.
Still, under the assumption that we just want `TupleTreeDiff`s as invalidation events, we already have a type-erased `TupleTreeDiff`, so we could discard the type erasure for invalidation event and adopt the type-erased `TupleTreeDiff`.
It'd simplify things a bit.

Reply:
I'm saying: all invalidation events are `TupleTreeInvalidationEvent`. So the first upcast is not necessary.

Reply:
but what is the point of that then? it just reduces flexibility, produces the same number of checks in client code except things have longer names. I would rather scrap events entirely and move around diffs, rather than have events that are just a wrapper.

Reply:
Well, my friend, that's not a bad idea at all!